File: archive/legacy_quarantine/apps/legacy_vault_android/app/src/main/java/com/sfx/legacyvault/core/models/ChatMessage.kt
package com.sfx.legacyvault.core.models

import java.util.UUID

data class ChatMessage(
    val id: String = UUID.randomUUID().toString(),
    val role: String = "user",
    val content: String = "",
    val timestamp: Long = 0L,
    val embeddingId: String? = null
)

File: archive/legacy_quarantine/apps/legacy_vault/legacy_vault/Core/Models/ChatMessage.swift
import Foundation

struct ChatMessage: Identifiable, Codable, Hashable {
    let id: UUID
    var role: MessageRole
    var content: String
    var timestamp: Date
    var embeddingId: UUID?

    enum MessageRole: String, Codable {
        case user, assistant
    }

        func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: ChatMessage, rhs: ChatMessage) -> Bool {
        lhs.id == rhs.id
    }
}
